Application/EdFi.Ods.Tests/EdFi.Ods.Sandbox/Repositories/ClientAppRepoTests.cs

```diff
@@ -23,30 +25,38 @@ public class When_calling_the_clientAppRepo
         {
             private ClientAppRepo _clientAppRepo;
             private ApiClient _testClient;
+            private DatabaseEngine _databaseEngine;
 
             [OneTimeSetUp]
             public void Setup()
             {
-                var configValueProviderStub = A.Fake<IConfigurationRoot>();
-                var usersContext = A.Fake<IUsersContext>();
+                var configuration = new ConfigurationBuilder()
+                    .SetBasePath(TestContext.CurrentContext.TestDirectory)
+                    .AddJsonFile("appsettings.json", false)
+                    .AddJsonFile("appsettings.Development.json", true)
+                    .AddEnvironmentVariables()
+                    .Build();
 
+                var engine = configuration.GetSection("ApiSettings").GetValue<string>("Engine");
 
-                var usersContextFactory = A.Fake<IUsersContextFactory>();
-                A.CallTo(() => usersContextFactory.CreateContext())
-                    .Returns(usersContext);
+                _databaseEngine = DatabaseEngine.TryParseEngine(engine);
 
-                _clientAppRepo = new ClientAppRepo(usersContextFactory, configValueProviderStub);
+                var connectionStringProvider = A.Fake<IAdminDatabaseConnectionStringProvider>();
+
+                A.CallTo(() => connectionStringProvider.GetConnectionString()).Returns(
+                    configuration.GetConnectionString("EdFi_Admin"));
+
+                var usersContextFactory = new UsersContextFactory(connectionStringProvider, _databaseEngine);
+
+                _clientAppRepo = new ClientAppRepo(usersContextFactory, configuration);
 
                 _testClient = new ApiClient(true)
                 {
                     Name = "ClientAppRepoTest" + Guid.NewGuid()
                         .ToString("N")
                 };
 
-                var dbSet = new TestDbSet<ApiClient> {_testClient};
-
-                A.CallTo(() => usersContext.Clients)
-                    .Returns(dbSet);
+                _clientAppRepo.UpdateClient(_testClient);
```

Comment:
You've turned this from a unit test to an integration test. At first I was going to ask you to delete this class altogether, since unit testing this repository is impractical.

However, after scrolling down some more, I remember now that we have a separate `EdFi.Admin.DataAccess.IntegrationTests` project. The `ClientAppRepo` class is part of `EdFi.Admin.DataAccess`, so this unit test never should have been under the `EdFi.Ods.Sandbox` in the first place.

Please keep your changes but move the file into project `EdFi.Admin.DataAccess.IntegrationTests`.